recorder/src/main/cpp/profile_writer.cc

```diff
@@ -259,7 +264,13 @@ ProfileSerializingWriter::ProfileSerializingWriter(jvmtiEnv* _jvmti, ProfileWrit
 }
 
 ProfileSerializingWriter::~ProfileSerializingWriter() {
-    if (cpu_samples_flush_ctr != 0) flush();
+    std::vector<PerfCtx::TracePt> user_ctxs;
+    reg.user_ctxs(user_ctxs);
+    auto last_ctx_id_reported = next_ctx_id;
```

Comment:
Maybe a different var name? last reported ctx id is next_ctx_id - 1. So maybe, var should be called candidate_ctx_id or value should be next_ctx_id - 1. Makes the usage in if conditional clearer.